src/black/linegen.py

```diff
@@ -197,6 +197,14 @@ def visit_decorators(self, node: Node) -> Iterator[Line]:
             yield from self.line()
             yield from self.visit(child)
 
+    def visit_power(self, node: Node) -> Iterator[Line]:
+        for idx, leaf in enumerate(node.children[:-1]):
+            next_leaf = node.children[idx + 1]
+            if leaf.type == token.NUMBER and next_leaf.type == syms.trailer:
```

Comment:
AFAIK `trailer` also includes the subscript and the call, e.g `1(duh)` or `1[duh]` would be also parenthesized (even though there is no such use case). I wonder whether we should make a more stricter check and ensure we are in an attribute trailer 🤔

Reply:
Sorry missed this, yeah I noticed that when I was going through the grammar file but subscript and calls on `token.NUMBER` isn't valid python syntax, so I left that case.

Reply:
It is actually a valid syntax, just not valid semantically. So we should probably handle it (check the first child of the trailer, if it is a dot, then it is an attribute access).

Reply:
@Shivansh-007 could you also add a test to verify this behavior? (both for subscript and calls)